crates/ide/src/folding_ranges.rs

```diff
@@ -264,6 +270,24 @@ fn fold_range_for_where_clause(where_clause: ast::WhereClause) -> Option<TextRan
     None
 }
 
+fn fold_range_for_multiline_match_arm(match_arm: ast::MatchArm) -> Option<TextRange> {
+    match match_arm.expr()?.syntax().kind() {
+        BLOCK_EXPR | MATCH_EXPR | ARRAY_EXPR | RECORD_EXPR => {
+            return None;
+        }
+        _ => {}
+    }
```

Comment:
I wonder if this should be
```suggestion
    if let Some(_) = fold_kind(match_arm.expr()?.syntax().kind()) {
        return None;
    }
```
Since if I see this right we don't want to emite folding ranges for things we already cover anyways

Reply:
I tried this, but cannot pass the test(test_fold_multiline_non_block_match_arm), for MATCH_EXPR and RECORD_EXPR it will generate matcharm fold which has block fold.
```
         match foo <fold block>{
                   qux2 => match foo2 <fold block>{
                        bar => (),
                    }</fold>,
                    strustS => StructS <fold block>{
                        a: 31,
                    }</fold>,
        }
```

Reply:
Oh, yes that is fine, that fold will still do the right thing and is more uniform so I think we can go with that.
I think I misunderstood you on your reply in the original issue then as I guess that was what you were asking I believe, sorry